mod-source-record-manager-server/src/main/java/org/folio/services/flowcontrol/RawRecordsFlowControlServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+package org.folio.services.flowcontrol;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.kafka.KafkaConsumerWrapper;
+import org.folio.verticle.consumers.consumerstorage.KafkaConsumersStorage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.folio.rest.jaxrs.model.DataImportEventTypes.DI_RAW_RECORDS_CHUNK_READ;
+
+@Service
+public class RawRecordsFlowControlServiceImpl implements RawRecordsFlowControlService {
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  @Value("${di.flow.max_simultaneous_records:100}")
+  private Integer maxSimultaneousRecords;
+  @Value("${di.flow.records_threshold:50}")
+  private Integer recordsThreshold;
+  @Value("${di.flow.control.enable:true}")
+  private boolean enableFlowControl;
+
+  @Autowired
+  private KafkaConsumersStorage consumersStorage;
+
+  private final AtomicInteger atomicCurrent = new AtomicInteger(0);
+
+  @PostConstruct
+  public void init() {
+    LOGGER.info("Flow control feature is {}", enableFlowControl ? "enabled" : "disabled");
+  }
+
+  @Override
+  public void trackChunkReceivedEvent(Integer initialRecordsCount) {
+    if (!enableFlowControl) {
+      return;
+    }
+
+    int current = atomicCurrent.addAndGet(initialRecordsCount);
+
+    LOGGER.info("--------------- Current value after chunk received: {} ---------------", current);
+
+    if (current >= maxSimultaneousRecords) {
+      List<KafkaConsumerWrapper<String, String>> rawRecordsReadConsumers = consumersStorage.getConsumersByEvent(DI_RAW_RECORDS_CHUNK_READ.value());
+
+      rawRecordsReadConsumers.forEach(consumer -> {
+        if (consumer.demand() > 0) {
+          consumer.pause();
```

Comment:
We don't need to handle exceptions here, I've added @Scheduled method that invoked each 5 min by default that will reset flow control state and back consumer to resume if it paused. It will help to avoid situation when consumer paused forever. More details you could see in java docs of resetState() method.